### PR TITLE
Networking for webots

### PR DESCRIPTION
**Pull request by CMurtagh-LGTM**

### CODE REVIEW

File: controllers/NUgus_Controller/tcp.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#ifndef CONTROLLERS_NUGUS_CONTROLLER_TCP
+#define CONTROLLERS_NUGUS_CONTROLLER_TCP
+
+#include <sys/socket.h>
+#include <vector>
+
```
Comment (KipHamiltons) [inline controllers/NUgus_Controller/tcp.hpp:6]: Are we going to put this in namespace `Controllers::NUgus_Controller`?

File: controllers/NUgus_Controller/tcp.cpp [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include "tcp.hpp"
+
+namespace tcp {
+
+}
```
Comment (KipHamiltons) [inline controllers/NUgus_Controller/tcp.cpp:5]: add new line please

File: controllers/NUgus_Controller/Makefile [hunk 3]
```diff
@@ -65,6 +65,9 @@
 ###
 ###-----------------------------------------------------------------------------
 
+CFLAGS = -Wall
```
Comment (KipHamiltons) [inline controllers/NUgus_Controller/Makefile:68]:
```suggestion
CFLAGS = -Wall -Wpedantic -Wextra -O3 -fPIC
CXXFLAGS = ${CFLAGS} -std=c++11
```
The standard is required for r-value references.

Not sure why we're using a makefile, but if we're setting flags here, these ones are good

**Review by KipHamiltons — COMMENTED** [review]
quick review

File: controllers/NUgus_Controller/Makefile (hunk 3 above)
Comment (ElderNoSpace) [reply controllers/NUgus_Controller/Makefile:68]: Webots controllers is made with the provided makefile.

Comment (Bidski) [reply controllers/NUgus_Controller/Makefile:68]: Probably better that we use cmake, maintain consistency across all of our repos as much as possible

Comment (ElderNoSpace) [reply controllers/NUgus_Controller/Makefile:68]: Is it possible to include a `makefile.include` in cmake?

Comment (Bidski) [reply controllers/NUgus_Controller/Makefile:68]: No, that is against the way cmake works

Comment (Bidski) [reply controllers/NUgus_Controller/Makefile:68]: gist.github.com › gsc07
Webots Controller build using CMake · GitHub

File: controllers/NUgus_Controller/tcp.hpp (hunk 1 above)
Comment (TrentHouliston) [reply controllers/NUgus_Controller/tcp.hpp:6]: That namespace makes me uncomfortable, what is it trying to be? camel case? snake case? a weird mix of the two?

Comment (Bidski) [reply controllers/NUgus_Controller/tcp.hpp:6]: I think we should aim for `controllers::nugus`

**Comment by CMurtagh-LGTM** [comment]
It looks like we don't have to make a controller. See [The api doc](https://cdn.robocup.org/hl/wp/2021/03/v-hsc_simulator_api.pdf)

**Comment by Bidski** [comment]
> It looks like we don't have to make a controller. See [The api doc](https://cdn.robocup.org/hl/wp/2021/03/v-hsc_simulator_api.pdf)

We will still need a controller, otherwise it will be extremely hard for us to test anything

**Review by CMurtagh-LGTM — COMMENTED** [review]
LGTM

File: .gitignore [hunk 4]
```diff
@@ -2,6 +2,6 @@ protos/**/*.cache
 worlds/.*.wbproj
 build/
 
-controllers/NUgus_Controller/NUgus_Controller
+/controllers/NUgus_Controller/NUgus_Controller
 
-nugus_controller_test
+/nugus_controller_test
```
Comment (CMurtagh-LGTM) [inline .gitignore]: Should this be a relative path?

Comment (Bidski) [reply .gitignore]: that file is in the root folder, so it shouldnt really matter

**Review by tayloryoung6396 — APPROVED** [review]
LGTM

File: controllers/nugus_controller/CMakeLists.txt [hunk 5]
```diff
@@ -1,14 +1,71 @@
...
+# Link against the webots target (this will also add any necessary include directories to our target)
+target_link_libraries(nugus_controller PRIVATE protobuf::libprotobuf-lite)
+
+# Generate binary in controller source directory
 set_target_properties(nugus_controller PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                                        OUTPUT_NAME nugus_controller)
+
+# ######################################################################################################################
+
+set(controller_test_sources NUgus_Client_Test.cpp "${CMAKE_CURRENT_BINARY_DIR}/protobuf/RobotControl.pb.cc"
```
Comment (ysims) [inline controllers/nugus_controller/CMakeLists.txt:52]: I'm getting an error on this line `Cannot find source file: NUgus_Client_Test.cpp`

Comment (Bidski) [reply controllers/nugus_controller/CMakeLists.txt]: Your branch is old

**Review by tayloryoung6396 — APPROVED** [review]
Again... it works for me

**Review by Bidski — APPROVED** [review]
LGTM to me too

**Pull request merged**